src/main/java/pro/trevor/tankgame/rule/impl/shared/TickRules.java

```diff
@@ -40,6 +43,19 @@ public class TickRules
         }
     );
 
+    public static <E extends Enum<E> & IAttribute> TickActionRule<GenericTank<E>> GetHealTanksInHealthPoolRule()
+    {
+        return new TickActionRule<>(
+            (s, t) -> {
+                if (t.getBoolean((E) TankAttribute.DEAD)) return;
```

Comment:
Why have this if statement here when it can be included as an additional condition in the below conditional?

Reply:
To clearly indicate an early return from the function. The below conditional expression is long enough in my opinion. Pre-conditions being checked early like this is common practice.